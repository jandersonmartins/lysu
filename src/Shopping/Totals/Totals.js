import React from 'react'
import PropTypes from 'prop-types'
import * as S from './Totals.styles'

const Totals = ({ label, value }) => (
  <S.Span>{label} <S.Strong>{value}</S.Strong></S.Span>
)

Totals.propTypes = {
  label: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ])
}

export default Totals
