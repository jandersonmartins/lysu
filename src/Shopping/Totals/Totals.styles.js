import styled from 'styled-components'

const Span = styled.span`
  font-size: 1.4rem;
`

const Strong = styled.strong`
  font-weight: 700;
`

export { Span, Strong }
