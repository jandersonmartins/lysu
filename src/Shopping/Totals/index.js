import Totals from './Totals'

export { Totals }
