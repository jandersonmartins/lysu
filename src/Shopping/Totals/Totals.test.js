import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import Totals from './Totals'

test('renders correctly', () => {
  const { container } = render(
    <ThemeProvider theme={{ colors: {} }}>
      <Totals label="Items" value="20" />
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})
