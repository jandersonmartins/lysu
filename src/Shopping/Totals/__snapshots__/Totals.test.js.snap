// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  font-size: 1.4rem;
}

.c1 {
  font-weight: 700;
}

<span
  class="c0"
>
  Items
   
  <strong
    class="c1"
  >
    20
  </strong>
</span>
`;
